python/lsst/daf/butler/cli/butler.py

```diff
@@ -44,19 +44,38 @@ def _initLogging(logLevel):
 
 
 def funcNameToCmdName(functionName):
-    """Change underscores, used in functions, to dashes, used in commands."""
+    """Convert function name to the butler command name: change underscores,
+    (used in functions) to dashes (used in commands), and change local-package
+    command names that conflict with python keywords to a leagal function name.
+    """
+    # The "import" command name and "butler_import" function name are defined
+    # in cli/cmd/commands.py, and if those names are changed they must be
+    # changed here as well.
+    # It is expected that there will be very few butler command names that need
+    # to be changed because of e.g. conflicts with python keywords (as is done
+    # here and in cmdNameToFuncName for the 'import' command). If this becomes
+    # a common need then some way of doing this should be invented that is
+    # better than hard coding the function names into these conversion
+    # functions. An extension of the 'cli/resources.yaml' file (as is currently
+    # used in obs_base) might be a good way to do it.
+    if functionName == "butler_import":
```

Comment:
I thought about making some kind of data structure like that, but if I'm going to do even that much work I wanted to take a moment to figure out the best way to solve the problem. I think it might involve the plugin system, but I didn't want to spent that much time on it or add complexity, yet.